frontend/svelte/src/lib/canisters/ic-management/ic-management.canister.ts

```diff
@@ -47,4 +48,40 @@ export class ICManagementCanister {
       throw mapError(e);
     }
   };
+
+  public updateSettings = async ({
+    canisterId,
+    settings,
+  }: {
+    canisterId: Principal;
+    settings: Partial<CanisterSettings>;
+  }): Promise<void> => {
+    try {
+      // Empty array does not change the value in the setting.
+      await this.service.update_settings({
+        canister_id: canisterId,
+        settings: {
+          controllers:
+            settings.controllers !== undefined
+              ? [settings.controllers.map((c) => Principal.fromText(c))]
```

Comment:
Just to spare a function
```suggestion
              ? [settings.controllers.map(Principal.fromText)]
```

Reply:
good point

Reply:
I couldn't, it needs to be called method style 😞 . This felt better than using `bind`.